### ISSUE TITLE
cmd/go: cgo test broke linux-amd64-nocgo build

### ISSUE THREAD

**Issue by bradfitz** [golang/go#26758]
The nocgo builder is broken as of https://go-review.googlesource.com/c/go/+/127297:

https://build.golang.org/log/9bced174cbb8cb8a6eeeca45926976b8989cb148

```
--- FAIL: TestScript (0.00s)
    --- FAIL: TestScript/mod_case_cgo (0.03s)
        script_test.go:150: 
            > env GO111MODULE=on
            > go get rsc.io/CGO
            [stderr]
            go: finding rsc.io/CGO v1.0.0
            go: downloading rsc.io/CGO v1.0.0
            can't load package: package rsc.io/CGO: build constraints exclude all Go files in $WORK/gopath/pkg/mod/rsc.io/!c!g!o@v1.0.0
            [exit status 1]
            FAIL: testdata/script/mod_case_cgo.txt:3: unexpected command failure
            
FAIL
FAIL	cmd/go	110.278s
```

Should be easy fix to exclude that test when cgo isn't available.

**Comment by mpx** [comment]
FYI, I'm still seeing failures with this test running all.bash at tip after the fix (b3b2f5d9) on a stock Fedora install:

```
go test proxy starting
go test proxy running at GOPROXY=http://127.0.0.1:37701/mod
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/language 14c0d48
go proxy: no archive golang.org/x/text/foo 14c0d48
go proxy: no archive sub.1 v1.0.0
go proxy: no archive badsub.1 v1.0.0
go proxy: no archive versioned.1 v1.0.0
go proxy: no archive versioned.1 v1.1.0
--- FAIL: TestScript (0.00s)
    --- FAIL: TestScript/mod_case_cgo (0.14s)
        script_test.go:150: 
            > [!cgo] skip
            > env GO111MODULE=on
            > go get rsc.io/CGO
            [stderr]
            go: finding rsc.io/CGO v1.0.0
            go: downloading rsc.io/CGO v1.0.0
            # rsc.io/CGO
            ccache: error: Failed to create directory /no-home/.ccache/tmp: Permission denied
            [exit status 2]
            FAIL: testdata/script/mod_case_cgo.txt:5: unexpected command failure
            
FAIL
FAIL	cmd/go	82.643s
```

Possibly similar to #17668.

**Comment by bradfitz** [comment]
@mpx, please file a new bug. This one is closed and yours looks unrelated.
